src/NHibernate/Proxy/NHibernateProxyBuilder.cs

```diff
@@ -74,6 +74,8 @@ public TypeInfo CreateProxyType(System.Type baseType, IReadOnlyCollection<System
 				interfaces.Add(baseType);
 			}
 
+			interfaces.RemoveWhere(i => !i.IsVisible);
+
```

Comment:
Shouldn't it emit a warning there?
It is not the dynamic proxy (which should be removed once lazy properties are handled by static proxies).

Something like "Interface {0} is not accessible, its members will not be proxified for entity {0}. If they access directly some internal state of the entity, unexpected outcome may occur.".

Due to the way NHibernate's proxies work, it could even be considered that trying to proxify entities implementing some internal interfaces is an error. But since the old proxies were not detecting the case and were letting silently these members unproxified, it would be a breaking change doable only for 6.0.

(It is an error because if some proxy instance is given back to some methods of the assembly holding the internal interfaces, it may call those methods. If they access directly internal or private state, they will then access the inherited proxy state instead of the delegated entity state handled by the proxy.)

Reply:
Well, the error case would also require, for actually being an error, to have these methods explicitly implemented. So unless this additional condition can be reliably detected, it should not be rejected as an error.

Reply:
I don't think we need any warnings here.

Reply:
I don't think that any warning necessary